index.html

```diff
@@ -156,9 +158,80 @@ <h2><dfn>SmartCardContext</dfn> interface</h2>
               optional sequence&lt;SmartCardProtocol&gt; preferredProtocols);
         };
       </pre>
+      <p>Instances of {{SmartCardContext}} are created with the internal slots
+      described in the following table:</p>
+      <table class="simple" data-dfn-for="SmartCardContext">
+        <tr>
+          <th>Internal slot
+          <th>Initial value
+          <th>Description (non-normative)
+        </tr>
+        <tr>
+          <td><dfn>[[\resourceManager]]</dfn></td>
+          <td>`null`</td>
+          <td>The platform's [[PCSC5]] `RESOURCEMANAGER` to be used.</td>
+        </tr>
+        <tr>
+          <td><dfn>[[\operationInProgress]]</dfn></td>
+          <td>`false`</td>
+          <td>Whether there is an ongoing PC/SC operation in this context.</td>
+        </tr>
+      </table>
       <section>
         <h3><dfn>listReaders()</dfn> method</h3>
-        <div class="issue">Write an algorithm for this method.</div>
+        <p>The {{SmartCardContext/listReaders()}} method steps
+        are:</p>
+        <ol>
+          <li>If [=this=].{{SmartCardContext/[[operationInProgress]]}} is
+            `true`, [=exception/throw=] a "{{InvalidStateError}}"
+            {{DOMException}}.</li>
```

Comment:
In other specifications I've tried to be consistent about either using exceptions or rejecting a promise but not both. I would rewrite this as:

```suggestion
          <li>Let |promise:Promise| be [=a new promise=].</li>
          <li>If [=this=].{{SmartCardContext/[[operationInProgress]]}} is
            `true`, [=reject=] |promise| with a "{{InvalidStateError}}"
            {{DOMException}} and return |promise|.</li>
```

Reply:
Done.